### ISSUE TITLE
Setting to disable new folder found notification

### ISSUE THREAD

**Issue by neomodern** [nextcloud/android#3080]
I use an app called Media Merger to automatically move pictures from various app folders into a central folder, and Nextcloud gives me a notification whenever it rediscovers these initial app folders when they briefly have a photo appear in them.

**Comment by bogie** [comment]
The same happens when nextcloud moves the pictures itself to subfolders

**Comment by rimeraz** [comment]
I'd like to disable this feature too. Every time I create a folder with images somewhere, Nextcloud client pops up with "new folder found". This is a bit annoying because I don't want to use this feature.

**Comment by danielb42** [comment]
I'd like to chime in on how silly it is, that the notification appears on auto-upload folders when NC is configured to delete or move processed pictures out of them..
But even without auto-upload, I tend to keep my default folders (such as "DCIM/Camera") clean and really don't need a notification every time I shoot a single pic or screenshot. If at least it wasn't a permanent notification..

**Comment by AndyScherzinger** [comment]
@danielb42 ouch... Yeah that is really bad UX, my suggestion would be to only show it once and remember that per notified folder so after you swiped it away for the folder it won't show up ever again for that folder. That should be done anyways.

What do you think @tobiasKaminsky?

**Comment by bogie** [comment]
@AndyScherzinger the notification also appears on folders which are already designated as auto upload.

To reproduce:
1. set camera folder as auto upload
2. have nextcloud move new photos/videos to its own directory when the auto-upload succeeds
3. the camera folder should be empty
4. take a picture
5. the notification will appear

**Comment by AndyScherzinger** [comment]
@bogie yeah that shouldn't happen either since the client has the info that this is a known and configured/active folder.

**Comment by Nutomic** [comment]
Same problem here. I had to disable notifications from Nextcloud as a workaround.

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky implemented a fix for the folder detection so a active folders shouldn't be notified and folders should be notified only once. Correct me if I am wrong. The fix is available via the dev build and is sheduled to shp with 3.4.0

**Comment by Nutomic** [comment]
I don't want any notifications like this, because I also use different tools to synchronize data to my phone. Even if I only get one notification per folder, that still means I get a useless notification every time I synchronize something new to my phone. If I want to synchronize something with Nextcloud, I will happily configure that myself.

**Comment by tobiasKaminsky** [comment]
Yes, it will be in 3.4.0.
@Nutomic you can also ony disable/mute the specific notification channel if you are on a recent android version.

This would be the ideal scenario for a "never again" button in notification.
Maybe someone finds the time?

**Comment by AndyScherzinger** [comment]
> This would be the ideal scenario for a "never again" button in notification.
Maybe someone finds the time?

How would that then be "reverted" if one ever decides to get these notifications again? Or is it a truely "never again - no way back folks" action?

**Comment by tobiasKaminsky** [comment]
Ha. Seems that I did not thought that well about it ;-)
We could put it in Settings->Notification, but cluttering settings is nothing I really like…

**Comment by AndyScherzinger** [comment]
Well, we could put it on the settings screen but only show the option in case it has been deactivated :)

**Comment by AndyScherzinger** [comment]
> This would be the ideal scenario for a "never again" button in notification.
Maybe someone finds the time?

@tobiasKaminsky DONE :) --> #3448

**Comment by dbielz** [comment]
Sorry, just found this issue. I also want to disable this feature, but completely. Not only the notifications. I manually choose the folders I want to have synced, and I don't see any sense in having to disable every single new found folder again, that Nextcloud added automatically only because a image was found.

**Comment by pquan** [comment]
Please, when did nextcloud turn into a data-gathering folder-scanning malware ? Yes, these are strong words, but we already have enough corporations (goog, fb, m$ etc) looking in our stuff and we don't need an "open source" corp-driven app to do the same.
It is ridiculous. I keep a library of books on my tablet (say a few thousands). Yes, books have an image. Can you imagine how many times I get a *beep* on reboot of the thing, or when eating, being in a meeting etc.
Yes, I can disable notifications for nextcloud, but I don't want it scanning my folders. Period. I sync what I want, the rest it mine. Thank you.

**Comment by AndyScherzinger** [comment]
> Please, when did nextcloud turn into a data-gathering folder-scanning malware ?

We don't, we just get notified by Android itself since Android is doing so no matter what the NExtcloud app does. So Google/Android scans your whole phone already. So with the latest change you can deactivate this notificatoin on older Android versions. The scanning is done by Android so you need to raise an issue with Google/Android not with this project.

**Comment by AndyScherzinger** [comment]
Okay further clarify and add some more details. Your description is quite close to the actual implementation except:
3. we don't add anything to a sync set. When opening the auto upload screen we ask Android itself for any kind of OS-known folder that contains media files (this is known as Android's media index), we then display all of these folders known to Androiod and check within the Nextcloud App's database which of these folders are known to the app and are active/configures (to tint the cloud icon blue) and do the right sorting.
Originally the list of folders is meant to give you an overview of actively monitored folders for auto upload (colored cloud icon) and any other folder that _could_ also work for auto upload (as-in has media files in it, think of it as some kind of a wizard/helper). 

So none of these folders that you never activated to be auto uploaded are watched by the app. They are just there to make it easier for people to add such a folder to the list of auto uploaded folders. There is another issue where the discussion is ongoing on which conditions should apply for a folder to actually show up on that list (e.g. more than one image within the folder, only show the top-level folder in case there are several subfodlers with images, things like that).

As seen in the other issue (e.g. #3263) we understand that having a large list on this screen is anyoing while we also need this list to work for people who just want to activate it for a list of folders (them being presented) and don't know or ever heard of "file system".

**Comment by zingmars** [comment]
Getting the same notification over and over again every time I connect a flash drive to my phone is becoming pretty annoying. Any actual ETA until the option to disable these notification makes it into the app?

**Comment by danielb42** [comment]
@zingmars disable the notification subtype in Androids' settings for the time being. With Samsungs UI there is Settings/Notifications/Nextcloud->File Monitor.

**Comment by dbielz** [comment]
> Okay further clarify and add some more details. Your description is quite close to the actual implementation except:
> 3. we don't add anything to a sync set. When opening the auto upload screen we ask Android itself for any kind of OS-known folder that contains media files (this is known as Android's media index), we then display all of these folders known to Androiod and check within the Nextcloud App's database which of these folders are known to the app and are active/configures (to tint the cloud icon blue) and do the right sorting.

Sorry for having be unclear. What I mean is this:
I have activated 1 or 2 certain folders for Autoupload. Now I try e.g. a new foto app, which does not save its images under 'DCIM/camera', but under, say 'fotoappX'. I shoot the first image, get Newclouds 'new folder found'-notification, remove this notification and continue testing. Later, on my PC, I find this new folder under 'SofortUpload/fotoappX' and it contains all images I just shot. And here I do not want new found folders to be activated automatically for autoupload.
Maybe meanwhile I do not read the 'new-folder-found"-notification not carefully enough, but the first times I saw it, I read it thoroughly and found no option to activate the new folder for autoupload or not. 
Still this always happens. 
While I find this notification a bit annoying (I know how to disable notifications totally if it disturbs too much), I absolutely don't like this activation of new folders. I always have to remove the activation manually afterwards.
I think there should be an extra option "Activate new found folders" or so, or to only activate subfolders of already active folders automatically. 

P.S.: I now tested again. If I don't simply remove the notification with 'remove all notifications' from android, I can tap on it and see the configuration dialog coming up. So maybe I have to research a bit more, why new folders are activated by default on my device. Maybe you have an idea.
Sorry for complaining.

**Comment by AndyScherzinger** [comment]
@dbielz thanks for the detailed description. _If_ you really have the issue that new folders get automatically activated for auto upload without you actually activating it then I'd ask you to open a separate issue for that because that would be a mayor bug and shouldn't happen under any circumstance!

No reason to be sorry. You are not complaining we just need some time and discussion to understand the real/root issue you are having here :)

**Comment by dbielz** [comment]
Thanks. In any case this discussion helped me to better understand how things work and where I have to look closer if thinks don't work as expected. I'll come back if this turns out to be a bug.
